examples/kubeflow-gke/main.py

```diff
@@ -43,43 +43,42 @@
-            tf.float32, shape=(BATCH_SIZE, mnist.IMAGE_PIXELS))
-        self.labels_placeholder = tf.placeholder(tf.int32, shape=(BATCH_SIZE))
+def train():
+    tf.logging.set_verbosity(tf.logging.ERROR)
+    data_sets = input_data.read_data_sets(INPUT_DATA_DIR)
+    images_placeholder = tf.placeholder(
+        tf.float32, shape=(BATCH_SIZE, mnist.IMAGE_PIXELS))
+    labels_placeholder = tf.placeholder(tf.int32, shape=(BATCH_SIZE))
 
-        logits = mnist.inference(self.images_placeholder,
-                                 HIDDEN_1,
-                                 HIDDEN_2)
+    logits = mnist.inference(images_placeholder,
+                             HIDDEN_1,
+                             HIDDEN_2)
 
-        self.loss = mnist.loss(logits, self.labels_placeholder)
-        self.train_op = mnist.training(self.loss, LEARNING_RATE)
-        self.summary = tf.summary.merge_all()
-        init = tf.global_variables_initializer()
-        saver = tf.train.Saver()
-        self.sess = tf.Session()
-        self.summary_writer = tf.summary.FileWriter(LOG_DIR, self.sess.graph)
-        self.sess.run(init)
+    loss = mnist.loss(logits, labels_placeholder)
+    train_op = mnist.training(loss, LEARNING_RATE)
+    summary = tf.summary.merge_all()
+    init = tf.global_variables_initializer()
+    saver = tf.train.Saver()
+    sess = tf.Session()
+    summary_writer = tf.summary.FileWriter(LOG_DIR, sess.graph)
+    sess.run(init)
 
-        data_set = self.data_sets.train
-        for step in xrange(MAX_STEPS):
-            images_feed, labels_feed = data_set.next_batch(BATCH_SIZE, False)
-            feed_dict = {
-                self.images_placeholder: images_feed,
-                self.labels_placeholder: labels_feed,
-            }
+    data_set = data_sets.train
+    for step in xrange(MAX_STEPS):
+        images_feed, labels_feed = data_set.next_batch(BATCH_SIZE, False)
+        feed_dict = {
+            images_placeholder: images_feed,
+            labels_placeholder: labels_feed,
+        }
 
-            _, loss_value = self.sess.run([self.train_op, self.loss],
-                                     feed_dict=feed_dict)
-            if step % 100 == 0:
-                print("At step {}, loss = {}".format(step, loss_value))
-                summary_str = self.sess.run(self.summary, feed_dict=feed_dict)
-                self.summary_writer.add_summary(summary_str, step)
-                self.summary_writer.flush()
+        _, loss_value = sess.run([train_op, loss],
+                                 feed_dict=feed_dict)
+        if step % 100 == 0:
+            print("At step {}, loss = {}".format(step, loss_value))
+            summary_str = sess.run(summary, feed_dict=feed_dict)
+            summary_writer.add_summary(summary_str, step)
+            summary_writer.flush()
 
 if __name__ == '__main__':
-    fairing.config.set_model(TensorflowModel())
-    fairing.config.run()
+    train = fairing.config.fn(train)
```

Comment:
I'd prefer to not have most of the examples depend on the function preprocessor and just work normally. Most times it will probably cause more issues than not

Reply:
I'd recommend function preprocessor for notebooks only. Only in notebooks there is a need to capture the current state of the objects so makes sense to use fn preprocessor's pickling feature. For python files, just simply executing the file remotely would be better without much restriction.

Reply:
I see two main way of people providing execution unit:
1. Users working in notebooks or ipython shells, create some code interactively and wants to execute that in the remote system. For this fn processor with cloudpickle is the right solution.
2. Users working in python files or notebooks, write code and except linear execution of it in the remote system. For this, base preprocessor or notebook preprocessor is the right solution.

thoughts?

Reply:
Sure, that's a reasonable separation of use cases. I'll work on updating the Python file examples to use alternative preprocessors, and keep the function preprocessor used in the notebook cases.